docs/config-file/v2.rst

```diff
@@ -0,0 +1,435 @@
+This means that the build stops at the first warning and exits with exit status 1.
+
+:Type: ``bool``
+:Default: ``false``
+
+submodules
+~~~~~~~~~~
+
+Git submodules configuration.
+
+.. note::
+   
+   You can't use ``include`` and ``exclude`` settings for submodules at the same time.
+
+Example:
+
+.. code-block:: yaml
+   
+   submodules:
+     include: all
+     recursive: true
+
+submodules.include
+``````````````````
+
+List of submodules to be included.
+
+:Type: ``list``
+:Default: ``[]``
+
+.. note::
+   
+   You can use the ``all`` keyword to include all submodules.
+
+   .. code-block:: yaml
+
+      submodules:
+        include: all
+
+submodules.exclude
+``````````````````
+
+List of submodules to be excluded.
+
+:Type: ``list``
+:Default: ``[]``
+
+.. note::
+   
+   You can use the ``all`` keyword to exclude all submodules.
+   This is the same as ``include: []``.
+
+   .. code-block:: yaml
+
+      submodules:
+        exclude: all
+
+submodules.recursive
+````````````````````
+
+Do a recursive clone of the submodules.
+
+:Type: ``bool``
+:Default: ``false``
+
+.. note::
+
+   This is ignored if there aren't submodules to clone.
+
+Schema
+------
+
+You can see the complete schema
+`here <https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/rtd_tests/fixtures/spec/v2/schema.yml>`_.
+
+Migrating from v1
+-----------------
+
+Changes
+~~~~~~~
+
+- The version setting is required. See :ref:`config-file/v2:version`.
+- The default value of the :ref:`config-file/v2:formats` setting has changed to ``[]``
+  and it doesn't include the values from the web interface.
+- The top setting ``requirements_file`` was moved to ``python.install``
+  and we don't try to find a requirements file if the option isn't present.
+  See :ref:`config-file/v2:Requirements file`.
+- The setting ``conda.file`` was renamed to ``conda.environment``.
+  See :ref:`config-file/v2:conda.environment`.
+- The ``build.image`` setting now only has two options: ``latest`` (default) and ``stable``.
+  See :ref:`config-file/v2:build.image`.
+- The settings ``python.setup_py_install`` and ``python.pip_install`` were replaced by ``python.install``.
+  And now it accepts a path to the package.
+  See :ref:`config-file/v2:Packages`.
+- The setting ``python.use_system_site_packages`` was renamed to ``python.system_packages``.
+  See :ref:`config-file/v2:python.system_packages`.
+- The build will fail if there are invalid keys (strict mode).
+
+.. warning::
+
+   Some values from the web interface are no longer respected,
+   please see :ref:`config-file/v2:Migrating from the web interface` if you have settings there.
+
+New
+~~~
+
+- :ref:`config-file/v2:sphinx`
+- :ref:`config-file/v2:mkdocs`
+- :ref:`config-file/v2:submodules`
+- :ref:`config-file/v2:python.install`
+
+Migrating from the web interface
+--------------------------------
+
+This should be pretty straightforward,
```

Comment:
Reading section is not too clear what I have to do as a user.

Reply:
Find the settings that match from the web interface